### ISSUE TITLE
`code_lowered` on specific method of `@generated` function should return CodeInfo for the expanded method body, not CodeInfo for the generator

### ISSUE THREAD

**Issue by jrevels** [JuliaLang/julia#22874]
Consider the following example:

```julia
julia> @generated function elmul(x, y::NTuple{N,Any}) where {N}
           return Expr(:tuple, [:(x * y[$i]) for i in 1:N]...)
       end
elmul (generic function with 1 method)

julia> @code_lowered elmul(1, (1,2,3))
CodeInfo(:(begin
        nothing
        SSAValue(0) = (Core.tuple)(:tuple)
        #7 = $(Expr(:new, :(Main.##7#8)))
        SSAValue(1) = #7
        SSAValue(2) = (Main.colon)(1, $(Expr(:static_parameter, 1)))
        SSAValue(3) = (Base.Generator)(SSAValue(1), SSAValue(2))
        SSAValue(4) = (Base.collect)(SSAValue(3))
        return (Core._apply)(Main.Expr, SSAValue(0), SSAValue(4))
    end))
```

This isn't unexpected, since generator expansion occurs during type inference, but it would be more useful if `@code_lowered` instead returned the  `CodeInfo` of the expanded method body for the given type signature. For example:

```julia
julia> @code_lowered elmul(1, (1,2,3))
CodeInfo(:(begin
        nothing
        return (Core.tuple)(x * (Main.getindex)(y, 1), x * (Main.getindex)(y, 2), x * (Main.getindex)(y, 3))
    end))
```

Motivation described in https://github.com/jrevels/Cassette.jl/issues/3 (under "Tracing Mechanism"), if anybody's curious.

**Comment by JeffBezanson** [comment]
Agreed. The problem is abstract types, for which we don't invoke generators. In those cases, we'd have to give an error (perhaps until we have fallback code for generated functions).

**Comment by yuyichao** [comment]
It should always be safe for `@code_lowered` though. For `code_lowered` I guess we can have arguments to control this (just like `code_llvm`) and pick a reasonable default behavior?

**Comment by vtjnash** [comment]
+1 to making `code_lowered` always return the unspecialized method body (also with a kwarg to optionally specify not to run the tfunc generator). I think we can make the generator itself accessed via something like `Core.Inference.code_tfunc`

**Comment by jrevels** [comment]
Fixed by #22979
